packages/contracts-bedrock/scripts/go-ffi/go-ffi.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+{
+    # Define a lock file
+    LOCKFILE="/tmp/go-ffi.lock"
+
+    # Function to clean up the lock file on exit
+    cleanup() {
+        rm -rf "$LOCKFILE"
+    }
+
+    # Get the directory of the script
+    dir=$(dirname "$(realpath "$0")")
+    # Check if the target binary exists
+    if [[ ! -f "$dir/$1" ]]; then
+        # Wait for the lock file to be released
+        while ! mkdir "$LOCKFILE" > /dev/null; do
+            sleep 0.3s
+        done
+
+        # Set a trap to call cleanup on exit
+        trap cleanup EXIT
+
+        # Ensure only one process is building the binaries
+        if [[ ! -f "$dir/$1" ]]; then
+            pushd "$dir/../.."
+            just build-go-ffi > /dev/null
+            popd
+        fi
```

Comment:
I find the nested if statements somewhat confusing. Is there a reason we can't just check for the binaries at the start of the script, and exit early if they already exist?

Reply:
Only the 1st process that grabs the file lock will enter this branch, we're using this condition to check whether it's the first or not, so that we don't need to store the result(whether first or not) in another bash variable.